EndlessClient/Input/UnwalkableTileActions.cs

```diff
@@ -74,26 +74,9 @@ public IReadOnlyList<UnwalkableTileAction> GetUnwalkableTileActions(IMapCellStat
             if (MainCharacter.RenderProperties.SitState != SitState.Standing)
                 return new[] { UnwalkableTileAction.None };
 
-            return cellState.Character.Match(
-                some: c => new[] { HandleWalkThroughOtherCharacter(c) }, //todo: walk through players after certain elapsed time (3-5sec?)
-                none: () => cellState.Warp.Match(
-                    some: w => new[] { HandleWalkToWarpTile(w), HandleWalkToTileSpec(cellState) },
-                    none: () => new[] { HandleWalkToTileSpec(cellState) }));
-        }
-
-        private UnwalkableTileAction HandleWalkThroughOtherCharacter(Character c)
-        {
-            // todo: caller handles this
-            //        EOGame.Instance.Hud.SetStatusLabel(EOResourceID.STATUS_LABEL_TYPE_ACTION,
-            //            EOResourceID.STATUS_LABEL_KEEP_MOVING_THROUGH_PLAYER);
-            //        if (_startWalkingThroughPlayerTime == null)
-            //            _startWalkingThroughPlayerTime = DateTime.Now;
-            //        else if ((DateTime.Now - _startWalkingThroughPlayerTime.Value).TotalSeconds > 5)
-            //        {
-            //            _startWalkingThroughPlayerTime = null;
-            //            goto case TileInfoReturnType.IsTileSpec;
-            //        }
-            return UnwalkableTileAction.Character;
+            return cellState.Warp.Match(
```

Comment:
Is there a reason you did this in `WalkValidationActions` rather than here in `UnwalkableTileActions`?

I think my plan for this feature was to `return UnwalkableTileActions.Character` here and have the `ArrowKeyController` handle it accordingly (start ghosting, walk through after X seconds). IMO handling ghosting doesn't really belong in `WalkValidationActions` because it's a bit beyond the scope of validating if a walk can happen

Reply:
> this could perhaps be in a better location around the arrow key handler or something perhaps

That's basically what I'm suggesting. Do the handling code at the `ArrowKeyController` level. WalkValidationActions doesn't really change, just indicates that a character is there, and if you get `Character` for more than {x} seconds the controller lets you pass through.

I haven't played around with this yet though - so maybe there's something I'm not thinking of as to why it doesn't work.

Reply:
This seems fine now that I've played around with it more. The way I've split the data vs methods out from the existing `GhostingRepository` it makes sense to keep this contained within `WalkValidationActions`.